### DISCUSSION TITLE
orderBy _count cant take where clause

### DISCUSSION THREAD

**Discussion by SuyashSingh15** [prisma/prisma#24315] [Q&A]
## Problem

Currently, if I try to sort users based on number of posts they have below is the query to be executed.
```js
const getUsers = await prisma.user.findMany({
  take: 10,
  orderBy: {
    posts: {
      _count: `'desc',`
    },
  },
})
```

But, if I want to sort users based on number of posts who have a postname as 'AdamCopeland', I seemingly cannot.

It would be better to have something like this ->
```js
const getUsers = await prisma.user.findMany({
  take: 10,
  orderBy: {
    posts: {
      _count: {
          where:{
              name : "AdamCopeland"
           },
           value:"desc"     //order of sort
       },
    },
  },
})
```

**Comment by ludralph** [comment]
Hi @SuyashSingh15 👋 

Thank you for raising this question.

This is not currently possible in Prisma Client. Can you please open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your use case?

**Comment by prkagrawal** [comment]
This would be a nice to have feature, you can also use a workaround like:
```javascript
async function getUsersSortedByPostName(postName: string, take: number) {
  const usersWithPostCount = await prisma.$queryRaw<
    { userId: number; postCount: number }[]
  >`
    SELECT "userId", COUNT(*) AS "postCount"
    FROM "Post"
    WHERE "name" = ${postName}
    GROUP BY "userId"
    ORDER BY "postCount" DESC
    LIMIT ${take}
  `;

  const userIds = usersWithPostCount.map(user => user.userId);

  const users = await prisma.user.findMany({
    where: {
      id: { in: userIds },
    },
    orderBy: {
      id: {
        _in: userIds, // Maintain the order of userIds
      },
    },
    take,
  });

  return users;
}

(async () => {
  const users = await getUsersSortedByPostName('AdamCopeland', 10);
  console.log(users);
})();
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
